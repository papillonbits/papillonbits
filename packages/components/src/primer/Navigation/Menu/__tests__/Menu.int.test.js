import { defaultItems as items, regular, withIcons, withHeading } from './Menu.int.story'

jest.mock('@papillonbits/library/array', () => ({
  getIndexItems: () => {},
  getIndexItemsWithSelected: () => {},
}))

jest.mock('@papillonbits/library/hooks', () => ({
  useState: () => {},
}))

const libraryArrayMockObject = require('@papillonbits/library/array')
const libraryHooksMockObject = require('@papillonbits/library/hooks')

describe('<Menu />', () => {
  let indexItemsDataObject = [
    { ...items[0], index: 0 },
    { ...items[1], index: 1 },
    { ...items[2], index: 2 },
    { ...items[3], index: 3 },
  ]

  const newIndexItemsWithSelectedDataObject = [
    { ...items[0], index: 0, isSelected: true },
    { ...items[1], index: 1, isSelected: false },
    { ...items[2], index: 2, isSelected: false },
    { ...items[3], index: 3, isSelected: false },
  ]

  const setIndexItemsMockFn = jest.fn((indexItems) => {
    indexItemsDataObject = indexItems
  })

  beforeEach(() => {
    jest.spyOn(libraryArrayMockObject, 'getIndexItems').mockReturnValue(indexItemsDataObject)

    jest.spyOn(libraryArrayMockObject, 'getIndexItemsWithSelected').mockReturnValue(newIndexItemsWithSelectedDataObject)

    jest.spyOn(libraryHooksMockObject, 'useState').mockImplementation(() => [indexItemsDataObject, setIndexItemsMockFn])
  })

  afterEach(() => jest.clearAllMocks())

  describe('Render', () => {
    test('must match regular()', () => {
      expect(global.renderToJSON(regular())).toMatchSnapshot()
    })

    test('must match withIcons()', () => {
      expect(global.renderToJSON(withIcons())).toMatchSnapshot()
    })

    test('must match withHeading()', () => {
      expect(global.renderToJSON(withHeading())).toMatchSnapshot()
    })
  })
})
