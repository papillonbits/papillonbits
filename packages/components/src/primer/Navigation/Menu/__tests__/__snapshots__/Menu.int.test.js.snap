// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Render must match regular() 1`] = `
<div>
  <nav
    aria-label="Person settings"
    aria-labelledby="menu-heading"
    class="menu"
  >
    <a
      aria-current="page"
      class="menu-item"
      href="#url"
    >
      Account
    </a>
    <a
      class="menu-item"
      href="#url"
    >
      Profile
    </a>
    <a
      class="menu-item"
      href="#url"
    >
      Emails
    </a>
    <a
      class="menu-item"
      href="#url"
    >
      Notifications
    </a>
  </nav>
</div>
`;

exports[`<Menu /> Render must match withHeading() 1`] = `
<div>
  <nav
    aria-label="Person settings"
    aria-labelledby="menu-heading"
    class="menu"
  >
    <span
      class="menu-heading"
    >
      Menu heading
    </span>
    <a
      aria-current="page"
      class="menu-item"
      href="#url"
    >
      Account
    </a>
    <a
      class="menu-item"
      href="#url"
    >
      Profile
    </a>
    <a
      class="menu-item"
      href="#url"
    >
      Emails
    </a>
    <a
      class="menu-item"
      href="#url"
    >
      Notifications
    </a>
  </nav>
</div>
`;

exports[`<Menu /> Render must match withIcons() 1`] = `
<div>
  <nav
    aria-label="Person settings"
    aria-labelledby="menu-heading"
    class="menu"
  >
    <a
      aria-current="page"
      class="menu-item"
      href="#url"
    >
      Account
    </a>
    <a
      class="menu-item"
      href="#url"
    >
      Profile
    </a>
    <a
      class="menu-item"
      href="#url"
    >
      Emails
    </a>
    <a
      class="menu-item"
      href="#url"
    >
      Notifications
    </a>
  </nav>
</div>
`;
